### ISSUE TITLE
typeVariablesWithNullableUpperBounds not working with java.util.function.Supplier

### ISSUE THREAD

**Issue by agrieve** [uber/NullAway#1198]
I'm trying to use a LibraryModel to make `Supplier<@Nullable Thing>` work as it should.

I've tried:

```java
    @Override
    public ImmutableSetMultimap<String, Integer> typeVariablesWithNullableUpperBounds() {
        return new ImmutableSetMultimap.Builder<String, Integer>()
                .put("java.util.function.Supplier", 0)
                .build();
    }

    @Override
    public ImmutableSet<String> nullMarkedClasses() {
        return ImmutableSet.of("java.util.function.Supplier");
    }
```

But cannot get the following to produce a warning:

```java
import java.util.function.Supplier;
import org.chromium.build.annotations.Nullable;
import org.chromium.build.annotations.NullMarked;

@NullMarked
class Foo {

  Supplier<@Nullable String> sup = () -> "";

  Foo() {
    sup.get().length();
  }
}
```

Running with `-XepOpt:NullAway:JSpecifyMode=true` and `-XepOpt:NullAway:OnlyNullMarked`

I also tested with a program-defined interface:
```
package test;

interface Bar<T> {
  T get();
}
```

And found that with the library model, I can get a warning to show if I run as:
```
javac.sh Bar.java Foo.java
```

But no warning shows when I switch the order of the sources:
```
javac.sh Foo.java Bar.java
```

(`javac.sh` is a helper to run javac with my library model .jar on the classpath)

**Comment by jeffrey-easyesi** [comment]
I think this is the same issue I saw - https://github.com/uber/NullAway/issues/1194
Try using `-XepOpt:NullAway:AnnotatedPackages=java.util.function.Supplier`

**Comment by msridhar** [comment]
@agrieve any way you can try a snapshot build of NullAway to see if this is already fixed on master?

**Comment by agrieve** [comment]
Happy to try, but I can't seem to find build instructions anywhere, and my guessing has so far proven unsuccessful.

It might be good to add "How to build locally" to the README.md or CONTRIBUTING.md...

I've tried `./gradlew shadowJar`, but it produces jars only for "jar-infer-cli" and "library-model-generator-cli". `./gradlew jar` builds a `nullaway.jar`, but it's missing its deps.

**Comment by msridhar** [comment]
Sorry, we should be better about docs on local builds; I will update them. 

Which jar files do you need?  Usually we do a local build by running `./gradlew publishToMavenLocal` which currently writes all the artifacts to `~/.m2/repository/com/uber/nullaway` under the `0.12.7-SNAPSHOT` folders.  We don't normally build and distribute a "NullAway with dependencies" jar ourselves, but maybe you can piece it together from that?

**Comment by agrieve** [comment]
Thanks, that got me unstuck, and indeed it's working!
